overlord/snapstate/handlers_link_test.go

```diff
@@ -1612,3 +1613,74 @@ func (s *linkSnapSuite) TestMaybeUndoRemodelBootChangesNeedsUndo(c *C) {
 	c.Check(s.stateBackend.restartRequested, HasLen, 1)
 	c.Check(s.stateBackend.restartRequested[0], Equals, state.RestartSystem)
 }
+
+func (s *linkSnapSuite) testDoLinkSnapWithToolingDependency(c *C, classicOrBase string, needsTooling bool) {
```

Comment:
nitpick you could get rid of the `needsTooling` parameter since we know it should be true if and only if `classicOrBase != "" && classicOrBase != "classic-system"` in the `default` case of the switch below.